src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/Razor/GenerateMethodCodeActionResolver.cs

```diff
@@ -119,15 +130,27 @@ public GenerateMethodCodeActionResolver(DocumentContextFactory documentContextFa
 
         var codeBehindTextDocEdit = new TextDocumentEdit()
         {
-            TextDocument = new OptionalVersionedTextDocumentIdentifier() { Uri = codeBehindUri },
+            TextDocument = codeBehindTextDocumentIdentifier,
             Edits = new TextEdit[] { edit }
         };
 
         return new WorkspaceEdit() { DocumentChanges = new[] { codeBehindTextDocEdit } };
     }
 
-    private WorkspaceEdit GenerateMethodInCodeBlock(RazorCodeDocument code, GenerateMethodCodeActionParams actionParams, string templateWithMethodSignature)
+    private async Task<WorkspaceEdit> GenerateMethodInCodeBlockAsync(
+        RazorCodeDocument code,
+        GenerateMethodCodeActionParams actionParams,
+        VersionedDocumentContext documentContext,
+        CancellationToken cancellationToken)
     {
+        var csharpSource = await documentContext.GetCSharpSourceTextAsync(cancellationToken).ConfigureAwait(false);
+        var templateWithMethodSignature = await PopulateMethodSignatureAsync(
+            documentContext,
+            actionParams,
+            csharpSource.ToString(),
```

Comment:
Can we avoid doing this? Isn't there logic already that works out where to put the generated method, that would already have a position? Or does that come later? Might be worth refactoring.